pkg/plugins/plugins.go

```diff
@@ -265,7 +266,7 @@ func Get(name, imp string) (*Plugin, error) {
 		logrus.Debugf("%s implements: %s", name, imp)
 		return pl, nil
 	}
-	return nil, ErrNotImplements
+	return nil, fmt.Errorf("%w: plugin=%q, requested implementation=%q", ErrNotImplements, name, imp)
```

Comment:
`errors.Wrapf` ?

Reply:
Do we actually _use_ `ErrNotImplements` anywhere? (except for in the test) https://github.com/moby/moby/blob/232ebf7fa469963e6cb9b43d66a953f5956ee5f3/pkg/plugins/plugin_test.go#L80

Should we use `errdefs.ErrNotImplemented` (and `errdefs.ErrNotFound` for `plugins.ErrNotFound`) ? https://github.com/moby/moby/blob/7d4b7883815d45a0b4e1f4c50ec566e24040cb2a/errdefs/defs.go#L46-L49

That way we could customise the error message ("plugin %s does not implement %s"), and would still be able to detect the type of error

Reply:
I don't think we can use packages from the repo in /pkg.
CI, as I recall, checks for this.

Reply:
And no, nothing that's vendored in used the error.

Reply:
btw, `%w` is go1.13's `errors.Wrap`